### ISSUE TITLE
Nice to have feature: skip sparse files (for deduplication)

### ISSUE THREAD

**Issue by fedy-cz** [sahib/rmlint#567]
I'm missing one (in my mind) pretty useful feature (option):
_To skip over all the sparse files during the search for duplicates._

**Use case:**
In many cases, sparse files get pre-allocated, and only once they are completely written (downloaded, generated, ...) they are immutable and it's safe to deduplicate them. While they are still incomplete (sparse) they can't be safely considered the same file (even if at bit level they currently are). Reflinks are safe, but not supported by many filesystems. Skipping sparse files and using hardlinks/symlinks should be a relatively safe workaround.

**Comment by jumperes** [comment]
Also:
 - The scan report currently displays apparent size for sparse files, even if less or no space will be freed.
 - `--size` filters by apparent size
 - All-empty sparse files apparently can't be cloned or reflinked. Running rmlint multiple times it repeatedly prints that it is cloning or reflinking (but also doesn't report any error).

On btrfs:
```
$ truncate -s 1M a b; sync; du a b
0	a
0	b
$ rmlint -gT df -c sh:handler=clone a b --size 4K
[...]
==> In total 2 files, whereof 1 are duplicates in 1 groups.
==> This equals 1.00 MB of duplicates which could be removed.
==> Scanning took in total 0.072s.
[...]
$ ./rmlint.sh -d
# ///This script will be deleted after it runs///
[  0%] Keeping:  <path-redacted>/a
[ 50%] Cloning to: <path-redacted>/b
[100%] Done!
Deleting script  ./rmlint.sh
$ rmlint -gT df -c sh:handler=clone a b
[...]
==> In total 2 files, whereof 1 are duplicates in 1 groups.
==> This equals 1.00 MB of duplicates which could be removed.
==> Scanning took in total 0.063s.
[...]
$ ./rmlint.sh -d
# ///This script will be deleted after it runs///
[  0%] Keeping:  <path-redacted>/a
[ 50%] Cloning to: <path-redacted>/b
[100%] Done!
Deleting script  ./rmlint.sh
```
